### DISCUSSION TITLE
如何在 Windows 系统从零开始构建一个和我一样的 Hugo 博客

### DISCUSSION THREAD

**Discussion by giscus** [ZexWoo/Giscus#2] [Announcements]
# 如何在 Windows 系统从零开始构建一个和我一样的 Hugo 博客

好兄弟的博客让我有点小羡慕，所以直接上手迁移。这篇文章本质上是流程备份，免得我以后忘了。菜鸡是这样的。

https://zexwoo.blog/posts/blogs/build-hugo-blog/

**Comment by ZexWoo** [comment]
测试一下评论功能好使不

**Comment by imagebear1** [comment]
根据博主的教程成功搭建了博客，非常感谢

**Comment by oddsunallday** [comment]
Eureka 和 Stack 个人认为确实是 Hugo 的两个不错的主题，能自定义的内容也不少，缺点是文档有些部分不太详细

**Reply by ZexWoo** [reply]
也没办法，只能靠自己再多多探索了。

**Comment by chojemmy** [comment]
教程写的真好，想尝试下Hugo，用stack主题一直有问题，搜到你这篇教程的时候已经没有力气再折腾了。回去继续乖乖用wordpress

**Reply by ZexWoo** [reply]
真遗憾。当然我也是摸索了挺久。主要就是图好看，还有多语言。

**Comment by chojemmy** [comment]
刚回复了一段评论没了，我用虚拟机重新装了个系统，真从零开始搞。到下载主题后 hugo server 启动报错了。不知道博主遇到没，应该咋解决？
ERROR 2023/03/18 20:52:31 Failed to get JSON resource "https://publish.twitter.com/oembed?dnt=false&omit_script=true&url=https%3A%2F%2Ftwitter.com%2FDesignReviewed%2Fstatus%2F1085870671291310081": Get "https://publish.twitter.com/oembed?dnt=false&omit_script=true&url=https%3A%2F%2Ftwitter.com%2FDesignReviewed%2Fstatus%2F1085870671291310081": dial tcp 108.160.170.52:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
If you feel that this should not be logged as an ERROR, you can ignore it by adding this to your site config:
ignoreErrors = ["error-remote-getjson"]
ERROR 2023/03/18 20:52:40 render of "home" failed: "C:\Blog\layouts\index.html:9:15": execute of template failed: template: index.html:9:15: executing "main" at <partial "article-list/default" .>: error calling partial: partial "article-list/default" timed out after 30s. This is most likely due to infinite recursion. If this is just a slow template, you can try to increase the 'timeout' config setting.
Error: Error building site: failed to render pages: render of "page" failed: "C:\Blog\layouts\_default\single.html:27:7": execute of template failed: template: _default/single.html:27:7: executing "main" at <partial "article/article.html" .>: error calling partial: "C:\Blog\layouts\partials\article\article.html:2:7": execute of template failed: template: partials/article/article.html:2:7: executing "partials/article/article.html" at <partial "article/components/header" .>: error calling partial: partial "article/components/header" timed out after 30s. This is most likely due to infinite recursion. If this is just a slow template, you can try to increase the 'timeout' config setting.
Built in 30134 ms

抱歉贴了一堆代码

**Reply by chojemmy** [reply]
解决了。第一个就直接注释掉错误，第二个竟然是生成页面的时间太久，超时了。
ignoreErrors: ["error-remote-getjson"]
timeout: 200000
在config里加上面两个就可以了。现在可以来修改主题了。感谢。
不过第一个应该要修改下主题哪个地方呢，不然会有资源在国外访问不了吧。

**Reply by ZexWoo** [reply]
应该没有国外的资源吧？你那个报错像是调用了推特的什么东西，你可以尝试去掉推特相关的账号信息一类的配置。

**Reply by chojemmy** [reply]
我F12看了网络，里面有https://fonts.googleapis.com的链接，有jsdelivr.net的，这些估计在国内都用不了，所以用时很久。

**Comment by chojemmy** [comment]
感谢博主，终于部署好了，明天整理一下。不过我看网络，估计是国外的主题，引用的资源有谷歌字体，有一些国外的js源。好慢啊，不晓得博主后面有做什么博客优化没？

**Reply by ZexWoo** [reply]
你是用自己的服务器还是 Vercel？我刚看了下我的，我也有一个 https://fonts.googleapis.com/ 的链接，但是没有降低加载速度，所以我之前没注意过这个问题。
Vercel 有个模块叫 Checkly，能够提供优化建议。

**Reply by chojemmy** [reply]
大概率是分区域的，有些地方（比如我这）基本就会加载很久，我有一个wordpress.com官方的二级博客，移动网络访问不了，但有网友反映说他们不用梯子也能正常访问。

**Reply by ZexWoo** [reply]
那就是网络问题了。

**Comment by chojemmy** [comment]
请教博主，这两步操作为啥不是在Hugo中启用 hugo-theme-stack 主题，而是直接把文件都复制出来。有什么区别吗？

跑完以后，进入 F:\Blog\themes\hugo-theme-stack 文件夹，将除了「exampleSite」和「config.yaml」以外的所有文件一股脑地复制粘贴到根目录，如果遇到说是否替换已有文件，就选「是」。

而后进入 F:\Blog\themes\hugo-theme-stack\exampleSite 文件夹，将「config.yaml」和「content」文件夹复制粘贴到根目录，同时删掉「config.toml」。

**Reply by ZexWoo** [reply]
我当时似乎是直接启用无效，被迫这样做的。

**Reply by ZexWoo** [reply]
你直接启用成功了是吗？

**Reply by chojemmy** [reply]
启用成功了啊，我还以为这一步操作有什么玄妙之处😦。
还有一个问题想请教一下。就是文章中的图片出现相册效果是怎么实现的，我也插入了类似的链接`[![](1.jpg)](1.jpg)`，但是点击图片就直接跳转了，并没有出现相册显示上下文图片。 唯一实现的就是把几张图片并排放置`![](1.jpg)![](2.jpg)![](3.jpg)`不插入文字，点击就能产生相册效果。到底应该怎么操作？

**Reply by chojemmy** [reply]
效果可以参照这个链接 [https://web.bim.to/p/emoji-support/](https://web.bim.to/p/emoji-support/)，前面产生相册效果的就是`![](1.jpg)![](2.jpg)![](3.jpg)`这样排列的，后面图片点击跳转就是`[![](1.jpg)](1.jpg)`这样写的markdown。但我看你的文章里图片明明是分开的，也产生了相册效果。

**Reply by ZexWoo** [reply]
我没有做任何特殊处理，也许是某个地方的配置，或是相关的 js 没有正确加载？

**Reply by ZexWoo** [reply]
我回顾了一下，看了一下主题制作者的教程，https://stack.jimmycai.com/guide/modify-theme 有这么一段说明：For example, in order to modify the themes/hugo-theme-stack/layouts/partials/header.html file, you will have to copy it to layouts/partials/header.html and modify it there (copy the code from theme's repository). The same applies to assets and static directories.

也就是说，如果复制出来，其优先级是高于 theme 文件夹内的主题文件的。所以，如果需要调整特定的部分，就把特定的文件复制出来就可以了，而其余的文件则保持原地不动。这样的话，可以随着作者来更新主题，而同时不丢失自己的改动。

全复制应该是当时偷懒的做法，我现在删除了一些不必要的文件。

**Comment by yukiime** [comment]
您好，可以请教一下web标签页前方的图标怎么设置吗，就是您标签页前方的图标Z

**Reply by ZexWoo** [reply]
在 `config.yaml` 中的 `params` 下属的 `favicon` 设置。

我在本地目录下没有找到事宜的路径，所以是采用了外部的图床 + 图链来实现的。

此处可以用 .ico 文件。

**Reply by yukiime** [reply]
感谢！

**Comment by candydre** [comment]
感谢，正准备搭博客，在犹豫WIX图片占比太大，而WP在移动端适配不好的时候，看到了这一篇优质作业，帮助很大。

**Comment by rickeygong** [comment]
求助！<br />
不知道为什么，我设置了中英文，在本地(hugo server)运行的时候，文章中英文切换没问题，我放到服务器后，中文->英文就不行，页面会跳转到localhost。
```yaml
languages:
    zh-cn:
        languageName: 中文
        title: Rickey
        languageCode: zh-cn
        weight: 1
        params:
            sidebar:
                subtitle: 我真诚地记录着，因为生活值得
    en:
        languageName: English
        title: Rickey
        weight: 2
        languageCode: en
        params:
            sidebar:
                subtitle: I sincerely document this because life is worth it.
```

**Reply by ZexWoo** [reply]
我估计，这不是双语问题，而是你 config 文件内没有指定自己准备好的域名，而是仍为默认的 localhost，导致本地无问题，部署后不正常。

**Reply by rickeygong** [reply]
非常感谢你的回复，我排查了很久，但是没法解决。<br />
我做过了尝试：我又重新部署了一遍，并用Stack主题的示例，使用hugo server 生成`public`，将`public`文件放到服务器，切换语言正常了，所以我确认应该是我自己调整的写代码有问题。<br />
你能帮我看下是哪里配置得有问题吗？<br />我的仓库：https://github.com/rickeygong/Gdh_Hugo_Blog<br />
Hugo 环境信息<br />
```sheel
$ hugo env

hugo v0.138.0-ad82998d54b3f9f8c2741b67356813b55b3134b9+extended windows/amd64 BuildDate=2024-11-06T11:22:34Z VendorInfo=gohugoio
GOOS="windows"
GOARCH="amd64"
GOVERSION="go1.23.2"
github.com/sass/libsass="3.6.6"
github.com/webmproject/libwebp="v1.3.2"
```

**Reply by rickeygong** [reply]
[已解决]<br />
网站部署到服务器时，应该使用`hugo`命令，而不是`hugo server`命令来生成静态文件
